client/components/AdoptionForm.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react"
+
+const AdoptionForm = () => {
+  const [adoptionData, setAdoptionData] = useState({
+    name: "",
+    phoneNumber: "",
+    email: "",
+    homeStatus: "own"
+  })
+  let errorContainer = null
+  const [errors, setErrors] = useState([])
+  if (errors.length !== 0) {
+    errorContainer = errors.map(error => {
+      return <p>{error}</p>
+    })
+  }
+
+  const handleChange = (event) => {
+    setAdoptionData({
+      ...adoptionData,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const checkForErrors = () => {
+    const errorsArray = []
+    const fieldsData = Object.keys(adoptionData)
+    fieldsData.forEach(fieldName => {
+      if (adoptionData[fieldName] === "") {
```

Comment:
While this check is generally ok, comparing the value to an empty string means that if the user supplies a bunch of spaces in the text field "      ", it will not be seen as an empty string. Take a look back at the article React Form Validations with Hooks for how we can trim strings and use lodash to check if any errors have been created!